R/data.R

```diff
@@ -52,11 +53,20 @@
 #' cat(readLines(file), sep = "\n")
 #'
 write_stan_json <- function(data, file) {
+  if (!is.list(data)) {
+    stop("'data' must be a list.", call. = FALSE)
+  }
   if (!is.character(file) || !nzchar(file)) {
     stop("The supplied filename is invalid!", call. = FALSE)
   }
 
   data_names <- names(data)
+  if (length(data) > 0 &&
+      (length(data_names) == 0 ||
+       length(data_names) != sum(nzchar(data_names)))) {
+    stop("All elements in 'data' list must have names.", call. = FALSE)
+
+  }
```

Comment:
not related to the "table" issue but we should check for this. this checks that each element in the data list has a name.